src/coreclr/src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -9215,32 +9275,70 @@ HRESULT ProfToEEInterfaceImpl::GetRuntimeInformation(USHORT * pClrInstanceId,
         return E_INVALIDARG;
     }
 
-    if (pcchVersionString != NULL)
+    if ((cchVersionString == 0) != (szVersionString == NULL))
     {
-        HRESULT hr = GetCORVersionInternal(szVersionString, (DWORD)cchVersionString, (DWORD *)pcchVersionString);
-        if (FAILED(hr))
-            return hr;
+        return E_INVALIDARG;
+    }
+
+    // defaults to use if the host hasn't set a version to parse
+    USHORT majorVersion = 0;
+    USHORT minorVersion = 0;
+    USHORT buildNumber = 0;
+
+    // Try and get the version set by the host
+    LPCWSTR fxProductVersion = Configuration::GetKnobStringValue(W("FX_PRODUCT_VERSION"));
```

Comment:
`FX_PRODUCT_VERSION` was thing when we have the parts in multiple different repos. Now that everything is building out of the same repo, I do not think you need that. We may be able to get rid of it.

Is this change actually changing anything now that we have bumped the versions of everything to 5.0? The `CLR_MAJOR_VERSION/CLR_MINOR_VERSION/CLR_BUILD_VERSION/CLR_BUILD_VERSION_QFE` should have the right numbers already.

Reply:
5.0.0-preview.4.20210.2 still returns 4.0.30319 for the version without these changes

Reply:
I would fix that instead of adding parsing the `FX_PRODUCT_VERSION` string. We should not have the `4.0.30319` version anywhere in the .NET 5 product.

Reply:
I thought about fixing the built in version, but my concern is if we are at preview 4 and it hasn't been fixed, it sounds like no one is using it. To fix it I would have to add some build logic that updates it, and it seems like the profiler would be the only consumer of it. So if it breaks or stops getting updated then likely no one else would notice so it could be broken for a while until a profiler vendor reports it.

I would like to use the same mechanism we use other places, and it seems like `FX_PRODUCT_VERSION` is what we are currently using for `Environment.Version` in C#.

Reply:
I have submitted https://github.com/dotnet/runtime/pull/35327 to delete `FX_PRODUCT_VERSION` since it is no longer necessary, and to cleanup the version macros - I believe that it should fix this bug as well.